service/src/main/kotlin/xyz/lebkuchenfm/external/storage/mongo/repositories/XSoundsMongoRepository.kt

```diff
@@ -60,6 +57,13 @@ class XSoundsMongoRepository(database: MongoDatabase) : XSoundsRepository {
     override suspend fun findByName(name: String): XSound? {
         return collection.find(eq(XSoundEntity::name.name, name)).firstOrNull()?.toDomain()
     }
+
+    override suspend fun incrementPlayCount(sound: XSound) {
```

Comment:
Let's return the updated `XSound?` here (just convert what `findOneAndUpdate` returns to domain object).